webapp-django/crashstats/status/context_processors.py

```diff
@@ -2,19 +2,22 @@
 
 
 def status_message(request):
-    statuses = StatusMessage.objects.filter(enabled=True)
 
-    if not statuses.exists():
-        return {}
+    def inner():
+        statuses = StatusMessage.objects.filter(enabled=True)
 
-    messages = []
-    for status in statuses.order_by('-created_at'):
-        messages.append({
-            'text': status.message,
-            'severity': status.severity,
-            'date': status.created_at,
-        })
+        if not statuses.exists():
+            return {}
+
+        messages = []
+        for status in statuses.order_by('-created_at'):
+            messages.append({
+                'text': status.message,
+                'severity': status.severity,
+                'date': status.created_at,
+            })
+        return messages
```

Comment:
In fact, we don't even need the `.values()`. It could just return a regular QuerySet. 
The idea of doing a closure is generally a good one because often not everything inside the context processor is as lazy as a QuerySet.